src/Rector/AbstractRector.php

```diff
@@ -50,17 +50,24 @@ public function setAbstractRectorDependencies(PropertyAdder $propertyAdder, Expr
      */
     final public function enterNode(Node $node)
     {
-        if (method_exists($this, 'getNodeType')) {
-            if (! is_a($node, $this->getNodeType(), true)) { // == basically "isCandidate()" condition
-                return null;
-            }
-        } elseif (method_exists($this, 'isCandidate')) {
-            if (! $this->isCandidate($node)) {
-                return null;
-            }
+        $nodeClass = get_class($node);
+
+        if (! in_array($nodeClass, $this->getNodeTypes(), true)) {
+            return null;
         }
+//        if (method_exists($this, 'getNodeTypes')) {
+//            if (! is_a($node, $this->getNodeTypes(), true)) { // == basically "isCandidate()" condition
+//                return null;
+//            }
+//        } elseif (method_exists($this, 'isCandidate')) {
+//            if (! $this->isCandidate($node)) {
+//                return null;
+//            }
+//        }
 
         $newNode = $this->refactor($node);
+
+        // nothing has changed, return original node
         if ($newNode !== null) {
```

Comment:
Wrong comment? If `$newNode` is different from null, it means we have a new Node